docker/Dockerfile

```diff
@@ -32,57 +38,64 @@ RUN groupadd comusers -g 9999 \
  && chown -R comuser:comusers /comsoftware \
  && chmod -R 6755 /comsoftware
 
-# Link version-specific aliases (python3 will be created later with virtual environment)
-RUN ln -s ~comuser/.venv/bin/python3 /usr/local/bin/python
-RUN ln -s /usr/bin/gfortran-12 /usr/bin/gfortran
-
 # all root steps completed above, now continue below as regular userID comuser
 USER comuser
 WORKDIR /home
 
 # Build NCEP libraries we need for SCM
 
-ENV NCEPLIBS_DIR /comsoftware/nceplibs
+ENV NCEPLIBS_DIR=/comsoftware/nceplibs
 
 RUN mkdir -p $NCEPLIBS_DIR/src && cd $NCEPLIBS_DIR/src \
   && git clone -b v2.4.1 --recursive https://github.com/NOAA-EMC/NCEPLIBS-bacio \
   && mkdir NCEPLIBS-bacio/build && cd NCEPLIBS-bacio/build \
   && cmake -DCMAKE_INSTALL_PREFIX=$NCEPLIBS_DIR .. \
-  && make VERBOSE=1 \
-  && make install 
+  && make VERBOSE=1 -j \
+  && make install
 
 RUN cd $NCEPLIBS_DIR/src \
   && git clone -b v2.3.3 --recursive https://github.com/NOAA-EMC/NCEPLIBS-sp \
   && mkdir NCEPLIBS-sp/build && cd NCEPLIBS-sp/build \
   && cmake -DCMAKE_INSTALL_PREFIX=$NCEPLIBS_DIR .. \
-  && make VERBOSE=1 \
+  && make VERBOSE=1 -j \
   && make install
 
 RUN cd $NCEPLIBS_DIR/src \
   && git clone -b v2.11.0 --recursive https://github.com/NOAA-EMC/NCEPLIBS-w3emc \
   && mkdir NCEPLIBS-w3emc/build && cd NCEPLIBS-w3emc/build \
   && cmake -DCMAKE_INSTALL_PREFIX=$NCEPLIBS_DIR .. \
-  && make VERBOSE=1 \
+  && make VERBOSE=1 -j \
   && make install
 
-ENV bacio_ROOT /comsoftware/nceplibs
-ENV sp_ROOT /comsoftware/nceplibs
-ENV w3emc_ROOT /comsoftware/nceplibs
+ENV bacio_ROOT=/comsoftware/nceplibs
+ENV sp_ROOT=/comsoftware/nceplibs
+ENV w3emc_ROOT=/comsoftware/nceplibs
+
+# Obtain CCPP SCM source code, build code, and download static data
+RUN if [ -z "$PR_NUMBER" ]; then \
+        cd /comsoftware \
+        && git clone --recursive -b main https://github.com/NCAR/ccpp-scm; \
+    else \
+        cd /comsoftware \
+        && git clone https://github.com/NCAR/ccpp-scm \
+        && cd ccpp-scm \
+        && git fetch origin pull/${PR_NUMBER}/head:test_pr \
+        && git checkout test_pr \
+        && git submodule update --init --recursive; \
+    fi
+
+RUN mkdir /comsoftware/ccpp-scm/scm/bin \
+  && cd /comsoftware/ccpp-scm/scm/bin \
+  && cmake ../src \
+  && make -j
 
-# Obtain CCPP SCM source code and static data, build code
-RUN cd /comsoftware \
-  && git clone --recursive -b main https://github.com/NCAR/ccpp-scm \
-  && cd /comsoftware/ccpp-scm/ \
+RUN cd /comsoftware/ccpp-scm/ \
   && ./contrib/get_all_static_data.sh \
   && ./contrib/get_thompson_tables.sh \
-  && cd /comsoftware/ccpp-scm/scm \
-  && mkdir bin \
-  && cd bin \
-  && cmake ../src \
-  && make -j4
+  && ./contrib/get_aerosol_climo.sh
```

Comment:
I think we shouldn't download this new aerosol data by default: it balloons the size of the container from 3.3 GB to a whopping 15.5 GB. We can update the instructions so users can download this data interactively if need be.

Actually now that I think about it, maybe it's best to have all the `contrib/` data downloaded separately from the container, using a bind mount to get that data into the container. That would reduce the whole container down to 1.9 GB. Do you think that makes sense?

Reply:
I was setting up the Docker file to download the repo, build it, and prepare everything to run. We could certainly slim down the Docker file to not download the `contrib/` data. Happy to set it up however we want it!